lib/instrumentation/aws-sdk/v3/redshift.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const { OperationSpec, QuerySpec } = require('../../../shim/specs')
+const InstrumentationDescriptor = require('../../../instrumentation-descriptor')
+const {
+  params: { DatastoreParameters }
+} = require('../../../shim/specs')
+const UNKNOWN = 'Unknown'
+
+function getRedshiftQuerySpec(shim, original, name, args) {
+  const [{ input }] = args
+
+  const isBatch = input.Sqls instanceof Array
+  const SqlQuery = isBatch ? input.Sqls[0] : input.Sql
+
+  return new QuerySpec({
+    query: SqlQuery,
+    parameters: setRedshiftParameters(this.endpoint, input),
+    callback: shim.LAST,
+    opaque: true,
+    promise: true
+  })
+}
+
+function getRedshiftOperationSpec(shim, original, name, args) {
+  const [{ input }] = args
+  return new OperationSpec({
+    name: this.commandName,
+    parameters: setRedshiftParameters(this.endpoint, input),
+    callback: shim.LAST,
+    opaque: true,
+    promise: true
+  })
+}
+
+async function getEndpoint(config) {
+  if (typeof config.endpoint === 'function') {
+    return await config.endpoint()
+  }
+
+  const region = await config.region()
+  return new URL(`https://redshift-data.${region}.amazonaws.com`)
```

Comment:
we talked on zoom. in newer versions of aws-sdk it's hitting this. our proposed entity relationship cannot make a link with just this but not related to instrumenting at the moment